### ISSUE TITLE
support `@base/object` module

### ISSUE THREAD

**Issue by PPpro** [cocos/cocos-engine#16375]
this should be done before the task https://github.com/cocos/cocos-engine/issues/16372

and it's a sub task of https://github.com/cocos/cocos-engine/issues/14293

**Comment by minggo** [comment]
Why make object itself be a separate module? Can't it be part of base module?

**Comment by PPpro** [comment]
> Why make object itself be a separate module? Can't it be part of base module?

as it's duscussed before https://github.com/cocos/cocos-engine/pull/16247#issue-1892133191, `base` should be a layer, and `object` should be a module in `base` layer

of course we can merge the total `base` layer as a full module in the future, but not now
